idmapping-rest/src/main/java/org/uniprot/api/idmapping/output/converter/uniprotkb/UniProtKBEntryPairFastaMessageConverter.java

```diff
@@ -17,7 +26,45 @@ public UniProtKBEntryPairFastaMessageConverter(Gatekeeper downloadGatekeeper) {
     }
 
     @Override
-    protected String toFasta(UniProtKBEntry entry) {
-        return UniProtKBFastaParser.toFasta(entry);
+    protected void before(
+            MessageConverterContext<UniProtKBEntryPair> context, OutputStream outputStream)
+            throws IOException {
+        TL_SUBSEQUENCE.set(context.isSubsequence());
+    }
+
+    @Override
+    protected String toFasta(UniProtKBEntryPair entryPair) {
+        if (TL_SUBSEQUENCE.get()) {
+            UniProtKBFasta uniProtKBFasta = getSubSequenceUniProtKBFasta(entryPair);
+            return UniProtKBFastaParser.toFasta(uniProtKBFasta);
+        } else {
+            return UniProtKBFastaParser.toFasta(entryPair.getTo());
+        }
+    }
+
+    @Override
+    protected void cleanUp() {
+        super.cleanUp();
+        TL_SUBSEQUENCE.remove();
+    }
+
+    private UniProtKBFasta getSubSequenceUniProtKBFasta(UniProtKBEntryPair entryPair) {
+        String from = entryPair.getFrom();
+        UniProtKBEntry entry = entryPair.getTo();
+        String range = from.substring(from.indexOf("[") + 1, from.indexOf("]"));
+
+        return new UniProtKBFastaBuilder()
+                .id(entry.getPrimaryAccession().getValue())
+                .entryType(entry.getEntryType())
+                .uniProtkbId(range)
+                .sequence(getSubSequence(range, entry.getSequence().getValue()))
+                .build();
+    }
+
+    private String getSubSequence(String range, String sequence) {
+        String[] rangeValues = range.split("-");
+        int sequenceStart = Integer.parseInt(rangeValues[0]);
```

Comment:
someone could send invalid request 
passing char instad of number
e-g instead of number 0 passing alphabet o